nbodykit/dask/test_tpm.py

```diff
@@ -0,0 +1,75 @@
+from nbodykit.extensionpoints import DataSource
+from binary_file import BinaryFile
+from fileiter import FileIterator
+from dask.delayed import delayed
+from dask import compute
+
+import numpy
+
+class DaskTPMSnapshot(DataSource):
+    """
+    DataSource to read snapshot files from Martin White's TPM simulations
+    """
+    plugin_name = "TPMSnapshot"
+    columns = ['Position', 'Velocity']
+    
+    def __init__(self, path, BoxSize, rsd=None, bunchsize=4*1024*1024):
+        
+        self.path = path
+        self.BoxSize = BoxSize
+        self.rsd = rsd
+        self.bunchsize = bunchsize
+        
+        dtypes = [('Position', ('f4', 3)), ('Velocity', ('f4', 3)), ('ID', 'u8')]
+        self.storage = BinaryFile(self.path, dtypes, header_size=28)
```

Comment:
Yes, I was looking at the existing stuff for Gadget and quickly realized my format was overly simplified. I'll stop by so we can chat more about what we want to do here...I was looking at how yt, etc handle Gadget, and I think at least found some data to test on.